.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,178 @@
+name: Regression tests
+
+on:
+  pull_request:
+    branches:
```

Comment:
This is not straightforward to implement, because we're using GitHub Actions PR contexts to get the right file changes to format and lint.

I've created [CPR-1650](https://linear.app/matterlabs/issue/CPR-1650/add-workflow-dispatch-trigger-for-llvm-regression-tests) to think about it later.